pallets/asset/src/lib.rs

```diff
@@ -471,6 +471,34 @@ decl_module! {
                 .map(drop)
         }
 
+        #[weight = <T as Config>::WeightInfo::create_asset(
+            name.len() as u32,
+            identifiers.len() as u32,
+            funding_round.as_ref().map_or(0, |name| name.len()) as u32
+        ) + <T as Config>::WeightInfo::register_custom_asset_type(custom_asset_type.len() as u32)]
+        pub fn create_asset_with_custom_type(
+            origin,
+            name: AssetName,
+            ticker: Ticker,
+            divisible: bool,
+            custom_asset_type: Vec<u8>,
+            identifiers: Vec<AssetIdentifier>,
+            funding_round: Option<FundingRoundName>,
+            disable_iu: bool,
+        ) -> DispatchResult {
+            let asset_type_id = Self::base_register_custom_asset_type(origin.clone(), custom_asset_type)?;
+            Self::base_create_asset(
```

Comment:
`Identity::<T>::ensure_perms(origin)?;` is checked twice (once for each of these calls).  The permissions check will only be checked against the main extrinsic `asset::create_asset_with_custom_type`, not for the two calls, since the `SignedExtensions` will not run for those two calls.

I think for the second call in these "batched" extrinsics, we use an `unsafe_*` function that does the work without the `ensure_perms` call.  The `base_*` function would just do the permissions check and call the `unsafe_*`.

Reply:
Good catch!